tools/src/download.c

```diff
@@ -2719,20 +2719,30 @@ int setRequestHeaders( HWTH_RETURNCODE_TYPE *rcPtr,
 char **getRequestHeaders() {
  char buf[80];
  char **headers = NULL;
- int NUM_HEADERS = 1;
+ int NUM_HEADERS = 2;
+
+ char *token = getenv("ZOPEN_GIT_OAUTH_TOKEN");
+ if (token == NULL) {
+	token = "";
+ }
 
 /***************************************************
  * Allocate an array of pointers to string, and
  * build a string for each of the headers.  Use a
  * NULL last array element to indicate end of array.
  ***************************************************/
  headers = (char **)calloc( 1+NUM_HEADERS, sizeof(char *) );
- sprintf( buf,
+ snprintf( buf, sizeof buf,
           "%s:%s",
           "User-Agent",
           "toolkit" );
- headers[0] = (char *)strdup(buf);
- headers[1] = NULL;
+ headers[0] = strdup(buf);
+
+ snprintf(buf, sizeof buf,"%s:%s %s",
```

Comment:
is this a valid header if token = "" ?

Reply:
It worked when I tested it, but I can rework the code to make it cleaner.  My excuse was that I was in a hurry to get it checked in.